### PR TITLE
Shorten various module names

### PR DESCRIPTION
**Pull request by michaelpj**
- Get rid of the `Language` prefixes
    - Particularly `Language.PlutusCore`, `Language.UntypedPlutusCore`, `Language.PlutusIR`, `Language.PlutusTx`, `Language.Plutus`
    - I spared Marlowe for now :)
- Shortened the `plutus-use-cases` module names

This is going to be disruptive, but I think it's better and we need to do it at some point.

Draft until I iron out the issues. I'll make sure to leave a commit that's just the renames so git tracks them properly.

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by kwxm** [comment]
I'm tempted to suggest `UntypedPlutusCore` -> `PlutusCore.Untyped` on the grounds that if you're using it in another package it might not be too obvious where to look for `UntypedPlutusCore`.

**Comment by kwxm** [comment]
I just noticed `plutus/plutus-core/plutus-core/src/Crypto.hs`, which I think may only be used in `Builtins.hs`.  There are a number of modules called `Crypto`, and there's `Crypto.Hash` as well, so this is a bit confusing.  Could we call it `PlutusCore.Crypto`, or maybe even amalgamate it with `Builtins.hs`?

File: notes/fomega/cek-cps-experiments/src/CekMachine.recursive.hs [hunk 1]
```diff
@@ -20,24 +20,24 @@
...
+import           PlutusCore.Constant
+import           PlutusCore.Evaluation.MachineException (MachineError (..), MachineException (..))
+import           PlutusCore.Evaluation.Result           (EvaluationResult (..))
+import           PlutusCore.View
 import           PlutusPrelude
 
-import qualified Data.Text                                       as T
-import qualified Language.PlutusCore.Pretty                      as PLC
+import qualified Data.Text                              as T
+import qualified PlutusCore.Pretty                      as PLC
```
Comment (kwxm) [inline notes/fomega/cek-cps-experiments/src/CekMachine.recursive.hs:37]: Shouldn't this be up above, with all the other `PlutusCore` things?

**Review by kwxm — APPROVED** [review]
This is so big that it's impossible to review!  My browser's having trouble handling the GitHub review page.  I did spot a couple of minor things as I was random scrolling though.

I find it a bit confusing now that we have `plutus-core/plutus-core` (from a previous PR): having two directories with the same name makes it easy to get lost or misplace things.  I don't have anything better to suggest though.

File: notes/fomega/cek-cps-experiments/src/CekMachine.recursive.hs (hunk 1 above)
Comment (michaelpj) [reply notes/fomega/cek-cps-experiments/src/CekMachine.recursive.hs:37]: I guess this stuff doesn't get formatted... I don't care too much about the stuff in `notes` tbh.

**Comment by michaelpj** [comment]
Also, I'm hoping this won't be too disruptive if git tracks the moves properly, this usually lets it DTRT for merges.

**Review by j-mueller — APPROVED** [review]

**Review by bezirg — APPROVED** [review]

**Comment by michaelpj** [comment]
Here we go!

**Comment by bezirg** [comment]
What about calling the current topmost `plutus-core` directory `plutus-lang` instead? Then we would have `plutus-lang/plutus-core`, `plutus-lang/plutus-ir`, etc

**Comment by michaelpj** [comment]
Well, it is all still about "Plutus Core" in its various forms, so it's not wildly off. I think rather we should move the `plutus-core` stuff inside it to `typed-plutus-core` or something.

**Pull request merged**